### PR TITLE
contribution guide updates

### PR DESCRIPTION
**Pull request by adamdubiel**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -2,7 +2,15 @@
 
 ## Picking up issues
 
-If you want to code something small in Hermes, search for `help wanted` tagged issues.
+If you want to code something small in Hermes, search for `ideal-for-contribution` tagged issues.
+
+## Commit message format
+
+```
+<issue number> | <commit message>
```
Comment (adamdubiel) [inline CONTRIBUTING.md]: this should make tools like waffle.io automatically link prs with issues

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -11,3 +19,10 @@ If you want to code something small in Hermes, search for `help wanted` tagged i
 * it would be nice to squash commits before creating pull requests
 * it's required to squash commits before merge
 * pay attention to changing documentation in `docs/`
+
+## Coding style
+
+* use `spock` when writing new unit tests in all modules
```
Comment (adamdubiel) [inline CONTRIBUTING.md]: let's leave junit behind already for new stuff

**Comment by druminski** [comment]
👍

**Comment by chemicL** [comment]
👍

**Pull request merged**